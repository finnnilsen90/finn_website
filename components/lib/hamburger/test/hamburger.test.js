const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer');
let { shallow, mount, render } = require('enzyme');

const Hamburger = require('../hamburger.js');

let newHamburger = new Hamburger();

let links_expected = {
    login: {
        link: '/logout',
        className: 'link',
        text: 'Logout'
    },
    home_page: {
        link: '/home_page',
        className: 'link',
        text: 'Home Page'
    },
    resume: {
        link: '/resume',
        className: 'link',
        text: 'Resume'
    },
    projects: {
        link: '/home_page',
        className: 'link',
        text: 'Projects'
    },
    contact: {
        link: '/contact',
        className: 'link',
        text: 'Contact'
    }
}

describe('component connection', () => {

    it('should be defined', () => {
        expect(Hamburger).toBeDefined();
    })

}) 

describe('<Hamburger />', () => {

    const wrapper = render(<Hamburger menu={links_expected}/>);

    it('renders components', () => {
        expect(wrapper).toBeTruthy();
    });

    it('renders an `.hamburger`', () => {
        expect(wrapper.find('.hamburger')).toHaveLength(1);
    });

});

describe('<Hamburger /> state', () => {

    it('should match state of each action', () => {
        const wrapper = shallow(<Hamburger menu={links_expected}/>);
        const hamburger = wrapper.find('.hamburger');
        const dropdown = wrapper.find('.dropdown');
        const container = wrapper.find('.libcomp-hamburger_Container');

        expect(wrapper).toMatchSnapshot();

        hamburger.simulate('onmouseover');
        expect(wrapper).toMatchSnapshot();

        dropdown.simulate('onmouseover');
        expect(wrapper).toMatchSnapshot();
        
    })

});


describe('<Hamburger /> dropdown state', () => {

    const wrapper = shallow(<Hamburger menu={links_expected}/>);
    const hamburger = wrapper.find('.hamburger');
    const dropdown = wrapper.find('.dropdown');

    let toggle = wrapper.state().toggle;

    let html =  [<li className="link" href="/logout">Logout</li>, <li className="link" href="/home_page">Home Page</li>, <li className="link" href="/resume">Resume</li>, <li className="link" href="/home_page"
>Projects</li>, <li className="link" href="/contact">Contact</li>]

    it('dropdown should snapshot at each state', () => {

        expect(dropdown).toMatchSnapshot();

        hamburger.simulate('onmouseenter');
        expect(dropdown).toMatchSnapshot();

        hamburger.simulate('onmouseleave');
        expect(dropdown).toMatchSnapshot();
    })

    it('html should match this array', ()=> {
        expect(newHamburger.links_compile(links_expected)).toEqual(html);
    })

});



describe('error reporting', () => {

    console.error = jest.fn();

    const wrapper = shallow(<Hamburger />);
    console.log('mockedError => ', console.error.mock.calls);

    it('should throw an error if no props passed', () => {
        expect(console.error).toHaveBeenCalledTimes(1);
    })
})