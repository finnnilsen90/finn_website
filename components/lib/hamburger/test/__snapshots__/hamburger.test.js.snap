// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> dropdown state dropdown should snapshot at each state 1`] = `
<ul
  className="dropdown"
  onMouseLeave={[Function]}
>
  <li
    className="link"
    href="/logout"
  >
    Logout
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Home Page
  </li>
  <li
    className="link"
    href="/resume"
  >
    Resume
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Projects
  </li>
  <li
    className="link"
    href="/contact"
  >
    Contact
  </li>
</ul>
`;

exports[`<Hamburger /> dropdown state dropdown should snapshot at each state 2`] = `
<ul
  className="dropdown"
  onMouseLeave={[Function]}
>
  <li
    className="link"
    href="/logout"
  >
    Logout
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Home Page
  </li>
  <li
    className="link"
    href="/resume"
  >
    Resume
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Projects
  </li>
  <li
    className="link"
    href="/contact"
  >
    Contact
  </li>
</ul>
`;

exports[`<Hamburger /> dropdown state dropdown should snapshot at each state 3`] = `
<ul
  className="dropdown"
  onMouseLeave={[Function]}
>
  <li
    className="link"
    href="/logout"
  >
    Logout
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Home Page
  </li>
  <li
    className="link"
    href="/resume"
  >
    Resume
  </li>
  <li
    className="link"
    href="/home_page"
  >
    Projects
  </li>
  <li
    className="link"
    href="/contact"
  >
    Contact
  </li>
</ul>
`;

exports[`<Hamburger /> state should match state of each action 1`] = `
<div
  className="libcomp-hamburger_Container"
>
  <div
    className="ham_container"
  >
    <div
      className="hamburger"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <hr />
      <hr />
      <hr />
    </div>
  </div>
  <div
    className="dropdown_container"
  >
    <ul
      className="dropdown"
      onMouseLeave={[Function]}
    >
      <li
        className="link"
        href="/logout"
      >
        Logout
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Home Page
      </li>
      <li
        className="link"
        href="/resume"
      >
        Resume
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Projects
      </li>
      <li
        className="link"
        href="/contact"
      >
        Contact
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Hamburger /> state should match state of each action 2`] = `
<div
  className="libcomp-hamburger_Container"
>
  <div
    className="ham_container"
  >
    <div
      className="hamburger"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <hr />
      <hr />
      <hr />
    </div>
  </div>
  <div
    className="dropdown_container"
  >
    <ul
      className="dropdown"
      onMouseLeave={[Function]}
    >
      <li
        className="link"
        href="/logout"
      >
        Logout
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Home Page
      </li>
      <li
        className="link"
        href="/resume"
      >
        Resume
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Projects
      </li>
      <li
        className="link"
        href="/contact"
      >
        Contact
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Hamburger /> state should match state of each action 3`] = `
<div
  className="libcomp-hamburger_Container"
>
  <div
    className="ham_container"
  >
    <div
      className="hamburger"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <hr />
      <hr />
      <hr />
    </div>
  </div>
  <div
    className="dropdown_container"
  >
    <ul
      className="dropdown"
      onMouseLeave={[Function]}
    >
      <li
        className="link"
        href="/logout"
      >
        Logout
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Home Page
      </li>
      <li
        className="link"
        href="/resume"
      >
        Resume
      </li>
      <li
        className="link"
        href="/home_page"
      >
        Projects
      </li>
      <li
        className="link"
        href="/contact"
      >
        Contact
      </li>
    </ul>
  </div>
</div>
`;
